packages/quantic/cypress/e2e/default-2/search-box/search-box.cypress.ts

```diff
@@ -0,0 +1,738 @@
+              Actions.pressEnterOnSearchBox(textarea);
+              const clickedSuggestionIndex = 0;
+              Expect.searchWithQuery(
+                exampleRecentQueries[clickedSuggestionIndex],
+                {
+                  LSkey: recentQueriesLSKey,
+                  queries: exampleRecentQueries,
+                }
+              );
+              Expect.logClickRecentQueries(
+                exampleRecentQueries[clickedSuggestionIndex]
+              );
+            });
+          });
+        });
+
+        describe('when using the keyboard to select a query suggestion', () => {
+          const exampleRecentQueries = ['foo', 'bar'];
+
+          beforeEach(() => {
+            setRecentQueriesInLocalStorage(exampleRecentQueries);
+            visitSearchBox({...defaultOptions, textarea});
+            mockQuerySuggestions(exampleQuerySuggestions);
+          });
+
+          it('should properly navigate and select the right suggestion', () => {
+            const expectedQuerySuggestions = [
+              ...exampleRecentQueries,
+              ...exampleQuerySuggestions,
+            ];
+            scope('when loading standalone search box', () => {
+              Expect.displaySearchBoxInput(true, textarea);
+              Expect.displaySearchButton(true);
+            });
+
+            scope('when focusing on the search box input', () => {
+              Actions.focusSearchBox(textarea);
+              cy.wait(InterceptAliases.QuerySuggestions);
+
+              Expect.displaySuggestionList(true);
+              Expect.displayClearRecentQueriesButton(true);
+              Expect.numberOfQuerySuggestions(expectedQuerySuggestions.length);
+              Expect.querySuggestionsEquals(expectedQuerySuggestions);
+            });
+
+            scope('when selecting a recent query with the keyboard', () => {
+              // First arrow down press to go to the clear recent queries option.
+              Actions.pressDownArrowOnSearchBox(textarea);
+              // pressing the arrow down button to surpass every recent query option.
+              for (let i = 0; i < exampleRecentQueries.length + 1; i++) {
+                Actions.pressDownArrowOnSearchBox(textarea);
+              }
+
+              Actions.pressEnterOnSearchBox(textarea);
+              const clickedSuggestionIndex = 0;
+              Expect.searchWithQuery(
+                exampleQuerySuggestions[clickedSuggestionIndex],
+                {
+                  LSkey: recentQueriesLSKey,
+                  queries: [
+                    exampleQuerySuggestions[clickedSuggestionIndex],
+                    ...exampleRecentQueries,
+                  ],
+                }
+              );
+              Expect.logClickSuggestion(
+                exampleQuerySuggestions[clickedSuggestionIndex]
+              );
+            });
+          });
+        });
+
+        describe('when a custom value is set for the property #numberOfSuggestions', () => {
```

Comment:
This should also be only a unit test?

Reply:
if we unit test the situations where we provide custom properties to the Quantic component every single Quantic component test should be updated x)

I see it tho, it's totally legit and makes sense  to unit test that, however the current logic we have with the Quantic tests goes with the opposite direction.

It's a good discussion to have tho, especially with the coming potential migration to playwright, we could take this occasion to adjust our logic to test Quantic components.

In my opinion, unit tests are the most essential tests to have, they are easier to have, robust and very stable.
Previously we were not preferring unit tests, because having always to mock headless was a pain, but we did that in several components in the C4SF and it was not that bad tbh, especially if we take some time make  this logic to mock headless more general and easy to use.

This way we could have much more unit tests and much less e2e tests, e2e tests will only focused on how a user interacts with the components in a search page and sending analytics and search requests.